src/views/dashboard/destination-details/filter/filter.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { useEffect, useState } from "react";
+import { useHistory, useParams } from "react-router-dom";
+import { useTranslation } from "react-i18next";
+import { Select } from "antd";
+import { Checkbox, Rate, DatePicker } from "antd";
+
+import { getFacilities } from "../../../../services/api/destination/facilityService";
+import { CAPACITY } from "../../../../shared/mocks/capacity.mock";
+import { FilterTranslation } from "../../context/filterTranslation";
+import filter from "../../../../assets/images/filter.svg";
+
+import "antd/dist/antd.css";
+import "./filter.scss";
+
+const Filter = () => {
+  const { Option } = Select;
+  const { RangePicker } = DatePicker;
+  const dateFormat = "DD-MM-YYYY";
+  const history = useHistory();
+  const params = useParams();
+  const { t: translate } = useTranslation();
+
+  const [hotelFacilities, setHotelFacilities] = useState<any[]>([]);
+  const [roomFacilities, setRoomFacilities] = useState<any[]>([]);
+  const [queryParams, setQueryParams] = useState<any>({
+    category: [],
+    roomFacilities: [],
+    accommodationFacilities: [],
+    capacity: 0,
+  });
+  const [filtersVisibility, setFiltersVisibility] = useState<boolean>(true);
+  const innerWidth = window.innerWidth;
+
+  useEffect(() => {
+    fetchHotelFacilities();
+    fetchRoomFacilities();
+  }, []);
+
+  useEffect(() => {
+    buildRoute();
+  }, [queryParams]);
+
+  useEffect(() => {
+    if (innerWidth < 480) setFiltersVisibility(false);
+  }, [innerWidth]);
+
+  const fetchHotelFacilities = async () => {
+    const { data } = await getFacilities("HOTEL");
+    setHotelFacilities(data);
+  };
+
+  const fetchRoomFacilities = async () => {
+    const { data } = await getFacilities("ROOM");
+    setRoomFacilities(data);
+  };
+
+  const buildRoute = () => {
+    history.push({
+      pathname: `/dashboard/destination/${params["name"]}`,
+      search: `?${
+        queryParams.capacity > 0 ? `capacity=${queryParams.capacity}` : ""
```

Comment:
As you see you are repeating yourself, inside utils function (as we talked above) you could have function that will go  though whole object and if there is something add it to string

Reply:
I can explain why this needs to be here